hedera-node/hedera-evm-api/src/main/java/com/hedera/services/evm/contracts/operations/HederaEvmStaticCallOperation.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (C) 2022 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hedera.services.evm.contracts.operations;
+
+import java.util.function.BiPredicate;
+import org.hyperledger.besu.datatypes.Address;
+import org.hyperledger.besu.evm.EVM;
+import org.hyperledger.besu.evm.frame.MessageFrame;
+import org.hyperledger.besu.evm.gascalculator.GasCalculator;
+import org.hyperledger.besu.evm.operation.StaticCallOperation;
+
+public class HederaEvmStaticCallOperation extends StaticCallOperation {
+    protected final BiPredicate<Address, MessageFrame> addressValidator;
+
+    public HederaEvmStaticCallOperation(
+            GasCalculator gasCalculator,
+            final BiPredicate<Address, MessageFrame> addressValidator) {
+        super(gasCalculator);
+        this.addressValidator = addressValidator;
+    }
+
+    @Override
+    public OperationResult execute(MessageFrame frame, EVM evm) {
+        return HederaEvmOperationsUtil.addressCheckExecution(
```

Comment:
Same Concern as HederaEvmStaticCallOperation.  If the "horizon" for historical re-execution is v0.33 we may not need this.